pkg/controller/direct/cloudbuild/workerpool_controller.go

```diff
@@ -135,10 +135,23 @@ func (m *model) AdapterForObject(ctx context.Context, reader client.Reader, u *u
 	if err != nil {
 		return nil, err
 	}
+
+	projectIDAndNumer := &refs.ProjectIDAndNumer{ID: projectRef.ProjectID}
+	network, found, err := unstructured.NestedString(u.Object, "status", "observedState", "networkConfig", "peeredNetwork")
+	if err == nil && found {
+		tokens := strings.Split(network, "/")
+		if len(tokens) == 5 && tokens[0] == "projects" {
+			projectIDAndNumer.Number = tokens[1]
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
 	return &Adapter{
-		id:        id,
-		gcpClient: gcpClient,
-		desired:   obj,
+		id:              id,
+		gcpClient:       gcpClient,
+		desired:         obj,
+		projectIDAndNum: projectIDAndNumer,
```

Comment:
This could get confusing because the project ID is the ID of the cloud build worker pool, but the project number is the number of the peered network

Reply:
Good point! Let me give a better solution. I think this project ID <--> Number deserves a more careful handling.

Reply:
I think the current `ProjectIDAndNumber` is not good. Mostly because I think we want to refactor both the `ProjectRef` and `ComputeNetwork` to a better format. Right now, none of the function can be reused.